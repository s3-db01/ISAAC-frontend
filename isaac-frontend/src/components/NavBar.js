import * as React from 'react';
import Box from '@mui/material/Box';
import Drawer from '@mui/material/Drawer';
import CssBaseline from '@mui/material/CssBaseline';
import Toolbar from '@mui/material/Toolbar';
import Divider from '@mui/material/Divider';
import SpeedIcon from '@mui/icons-material/Speed';
import ThermostatIcon from '@mui/icons-material/Thermostat';
import NotificationsIcon from '@mui/icons-material/Notifications';
import SettingsInputCompositeIcon from'@mui/icons-material/SettingsInputComposite';
import SettingsIcon from '@mui/icons-material/Settings';

import {Tab, Tabs} from '@material-ui/core/';
import {
	Route,
	Link,
} from 'react-router-dom';

import logo from '../images/isaac-primary-logo-rgb-500px.png';

const drawerWidth = 240;


// app bar
export default function PermanentDrawerLeft() {
	const styles = {
		toolbarContainer: {
			backgroundImage: `url(${logo})`,
			backgroundSize: 'cover',
			backgroundRepeat: 'no-repeat',
			backgroundPosition: 'center center',
			height: 100,
		},
	};

	return (
		<Box sx={{display: 'flex'}}>
			<CssBaseline />
			<Drawer
				sx={{
					'width': drawerWidth,
					'flexShrink': 0,
					'& .MuiDrawer-paper': {
						width: drawerWidth,
						boxSizing: 'border-box',
					},
				}}
				variant="permanent"
				anchor="left"
			>
				<Toolbar style={styles.toolbarContainer}/>
				<Divider />
				<Route
					path='/'
					render={(history) => (
						<Tabs
							orientation="vertical"
							value={history.location.pathname}
							textColor="secondary"
							indicatorColor="secondary"
						>
							<Tab
								label={<div style={{fontFamily: 'Open Sans'}}>Dashboard</div>}
								value="/"
								component={Link}
								to="/"
								icon={<SpeedIcon/>}
							/>
							<Tab
								label={<div style={{fontFamily: 'Open Sans'}}>Heatmap</div>}
								value="/heatmap"
								component={Link}
								to="/heatmap"
								icon={<ThermostatIcon/>}
							/>
							<Tab
								label={<div style={{fontFamily: 'Open Sans'}}>Notifications</div>}
								value="/notifications"
								component={Link}
								to="/notifications"
								icon={<NotificationsIcon/>}
							/>
							<Tab
								label={<div style={{fontFamily: 'Open Sans'}}>Advanced</div>}
								value="/advanced"
								component={Link}
								to="/advanced"
								icon={<SettingsInputCompositeIcon />}
							/>
							<Tab
								label={<div style={{fontFamily: 'Open Sans'}}>Settings</div>}
								value="/settings"
								component={Link}
								to="/settings"
								icon={<SettingsIcon/>}
							/>
						</Tabs>
					)}>
				</Route>
				<Divider />
			</Drawer>
			<Box
				component="main"
				sx={{flexGrow: 1, bgcolor: 'background.default', p: 3}}
			>
				<Toolbar />
			</Box>
		</Box>
	);
}
