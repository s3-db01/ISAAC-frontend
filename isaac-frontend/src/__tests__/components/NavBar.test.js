import React from 'react';
import NavBar from '../../components/NavBar';
import { shallow, render, mount, configure } from 'enzyme';
import toJSON from 'enzyme-to-json';

test('should render NavBar without crashing', () => {
	shallow(<NavBar />);
});

test('should render NavBar correctly', () => {
	const tree = shallow(<NavBar />);
	expect(toJSON(tree)).toMatchSnapshot();
});


