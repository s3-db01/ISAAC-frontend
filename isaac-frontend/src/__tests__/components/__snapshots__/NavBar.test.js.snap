// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar correctly 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "display": "flex",
    }
  }
>
  <CssBaseline />
  <ForwardRef(Drawer)
    anchor="left"
    sx={
      Object {
        "& .MuiDrawer-paper": Object {
          "boxSizing": "border-box",
          "width": 240,
        },
        "flexShrink": 0,
        "width": 240,
      }
    }
    variant="permanent"
  >
    <ForwardRef(Toolbar)
      style={
        Object {
          "backgroundImage": "url(test-file-stub)",
          "backgroundPosition": "center center",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
          "height": 100,
        }
      }
    />
    <ForwardRef(Divider) />
    <Route
      path="/"
      render={[Function]}
    />
    <ForwardRef(Divider) />
  </ForwardRef(Drawer)>
  <ForwardRef(Box)
    component="main"
    sx={
      Object {
        "bgcolor": "background.default",
        "flexGrow": 1,
        "p": 3,
      }
    }
  >
    <ForwardRef(Toolbar) />
  </ForwardRef(Box)>
</ForwardRef(Box)>
`;
